detox/src/realms/DetoxPrimaryContext.test.js

```diff
@@ -345,96 +357,73 @@ describe('DetoxPrimaryContext', () => {
 
         expect(globalLifecycleHandler.emergencyCleanup).not.toHaveBeenCalled();
         expect(detoxServer().close).not.toHaveBeenCalled();
-        expect(ipcServer.dispose).not.toHaveBeenCalled();
+        expect(ipcServer().dispose).not.toHaveBeenCalled();
       });
     });
   });
 
 
   function _initDetoxConfig() {
-    detoxConfigDriver = new DetoxConfigDriver(_.cloneDeep(detoxConfigBase));
+    detoxConfigDriver = new DetoxConfigDriver(_.cloneDeep(DETOX_CONFIG_BASE));
 
     jest.mock('../configuration');
-    configuration = require('../configuration');
+    configuration = jest.requireMock('../configuration');
     configuration.composeDetoxConfig.mockResolvedValue(detoxConfigDriver.detoxConfig);
   }
 
   function _initExternalMocks() {
     jest.mock('signal-exit');
+    signalExit = jest.requireMock('signal-exit');
     jest.mock('fs-extra');
-    fs = require('fs-extra');
+    fs = jest.requireMock('fs-extra');
```

Comment:
I'm using `jest.requireMock` a lot to make sure that the typings match.

Reply:
Could you elaborate?

Reply:
In TypeScript, `jest.requireMock` returns `Mock<T>` which plays well with the JSDoc `/** @type {} */` declarations I define above. Altogether, it gives us better type-checking without the need to `@ts-ignore`.

Reply:
Awesome 😎